snapshots/devmapper/config.go

```diff
@@ -105,5 +112,13 @@ func (c *Config) Validate() error {
 		result = multierror.Append(result, fmt.Errorf("base_image_size is required"))
 	}
 
+	if c.FileSystemType != "" {
```

Comment:
should remove this if-condition because the `parse()` is called before `Validate()`.
The `Validate()` should not ignore the empty case.

Reply:
I agree with you. However if you look at the parse() function, its actually setting fsTypeExt4 for empty cases. So technically we will never have empty case inside validate().  I can move the empty handling to validate() and remove it from parse(). But it will be the same thing. Do you think we still need a change?

Reply:
Sorry for late reply. Yeah, technically, the `FileSystemType` will never be empty. But for the `Validate()`, the value should be validate. If we skip the empty case, someone changes the `parse()` and might hit the bug here. So I think we should change `Validate()` like:

```go
func (c *Config) Validate() error {
...
                switch c.FileSystemType {
		case fsTypeExt4, fsTypeXFS:
		default:
			result = multierror.Append(result, fmt.Errorf("unsupported Filesystem Type: %q", c.FileSystemType))
		}
      ...
}
```

Sounds good to you?